File: backends/ts-hono/src/domain/staff/enums.ts
/**
 * スタッフドメイン 列挙定数モジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */

/** スタッフロール定数。 */
export const StaffRole = {
  MEMBER: 0,
  ADMIN: 1,
} as const;

export type StaffRoleValue = typeof StaffRole[keyof typeof StaffRole];

/** OAuth プロバイダー定数。 */
export const Provider = {
  GOOGLE: 1,
  GITHUB: 2,
} as const;

export type ProviderValue = typeof Provider[keyof typeof Provider];

File: backends/rust-axum/src/domain/client/enums.rs
//! クライアントドメイン 列挙定数モジュール。
//!
//! # Author
//! Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

/// クライアントステータス: 非アクティブ。
pub const STATUS_INACTIVE:  i32 = 1;
/// クライアントステータス: 稼働中。
pub const STATUS_ACTIVE:    i32 = 2;
/// クライアントステータス: 停止中。
pub const STATUS_SUSPENDED: i32 = 3;
/// クライアントステータス: 閉鎖。
pub const STATUS_CLOSED:    i32 = 4;
